docs/en/how_to_guides/finetune_with_a_custom_dataset.md

```diff
@@ -0,0 +1,357 @@
+
+```bash
+python finetune.py --config=./configs/mobilenetv3/mobilnet_v3_small_ascend.yaml --data_dir=./aircraft/data --pretrained=True
+```
+
+When fine-tuning mobilenet v3-small based on Aircraft dataset, this tutorial mainly made the following changes to the hyper-parameters:
+
+| Hyper-parameter | Before     | After                |
+| --------------- | ---------- | -------------------- |
+| dataset         | "imagenet" | ""                   |
+| batch_size      | 75         | 8                    |
+| image_resize    | 224        | 600                  |
+| auto_augment    | -          | "randaug-m7-mstd0.5" |
+| num_classes     | 1000       | 100                  |
+| pretrained      | False      | True                 |
+| epoch_size      | 470        | 50                   |
+| lr              | 0.77       | 0.002                |
+
+### Fine-tuning All the Parameters
+
+Since the progress of this type of fine-tuning is the same as training from scratch, simply **start the training by running finetune.py ** and adjust the parameters as training from scratch.
+
+### Freeze Feature Network
+
+#### Linear Probe
+
+We prevent parameters from updating by setting `requires_grad=False` for all parameters except those in the full connection layer. In finetune.py, add the following code after `create_model` :
+
+```python
+from mindcv.models.registry import _model_pretrained_cfgs
+
+# ...create_model()
+
+# number of parameters to be updated
+num_params = 2
+
+# read names of parameters in FC layer
+classifier_names = [_model_pretrained_cfgs[args.model]["classifier"] + ".weight",
+                    _model_pretrained_cfgs[args.model]["classifier"] + ".bias"]
+
+# prevent parameters in network(except the classifier) from updating
+for param in network.trainable_params():
+    if param.name not in classifier_names:
+        param.requires_grad = False
+```
+
+#### Freeze Part of the Feature Network
+
+To balance the speed and precision of fine-tuning, we can also fix some target network parameters and train the parameters in the deep layer only. It is necessary to extract the parameter names in those layers to be frozen and slightly modify the code in the last chapter. By printing the result of `create_model` -- `network`, we can see that in MindCV, each layer of the network of mobilenet v3-small is named with `features.*`. Suppose that we freeze only the first 7 layers of the network, add the following code after `create_model`:
+
+```python
+# ...create_model()
+
+# read names of network layers
+freeze_layer=["features."+str(i) for i in range(7)]
+
+# prevent parameters in the first 7 layers of the network from updating
+for param in network.trainable_params():
+    for layer in freeze_layer:
+    	if layer in param.name:
+        	param.requires_grad = False
+```
```

Comment:
Aren't these params already frozen by the code in #### Linear Probe ?

Reply:
This is the example code of `Freeze Part of the Feature Network`, which is another circumstance different from `Linear Probe`.